src/StreamJsonRpc/HeaderDelimitedMessageHandler.cs

```diff
@@ -233,12 +233,18 @@ private enum HeaderParseState
             sendingBufferStream.Write(CrlfBytes, 0, CrlfBytes.Length);
 #pragma warning restore VSTHRD103 // Call async methods when in an async method
 
+            // Either write both the header and the message, or don't write anything.
+            // Fix for https://github.com/Microsoft/vs-streamjsonrpc/issues/55.
```

Comment:
I'd rather include details like the issue being fixed in the git commit comments. The code comments should be sufficiently explanatory on their own to help the user understand why the code is written the way it is without relying on context in a bug that is hopefully still accessible. Sometimes when the issue is particularly complex, or the bug is still active and the comment is on a disabled test, I'll go ahead and put the link to the bug in the code, but otherwise I like to avoid it.